components/OddsCalculatorDialog/InvestigatorOddsView.tsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+
+import Card from '../../data/Card';
+import typography from '../../styles/typography';
+import { ChaosBag, ChaosTokenType, SPECIAL_TOKENS } from '../../constants';
+import { COLORS } from '../../styles/colors';
+import SkillOddsRow from './SkillOddsRow';
+import { CampaignDifficulty } from '../../actions/types';
+
+export interface InvestigatorOddsProps {
+  chaosBag: ChaosBag;
+  currentScenarioCard: Card;
+  difficulty?: string;
+  investigator: Card;
+  testDifficulty: number;
+}
+
+type Props = InvestigatorOddsProps;
+
+interface State {
+  specialTokenValues: { token: ChaosTokenType; value: number; raw_value: string | null }[];
+}
+
+export default class InvestigatorOddsView extends React.Component<Props, State> {
+
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      specialTokenValues: [],
+    };
+  }
+
+  componentDidMount(): void {
+    this.getSpecialTokenValues();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (this.props.currentScenarioCard !== prevProps.currentScenarioCard) {
+      this.getSpecialTokenValues();
+    }
+  }
+
+  getSpecialTokenValues() {
+    const {
+      currentScenarioCard,
+      difficulty,
+    } = this.props;
+    const scenarioTokens:
+      { token: ChaosTokenType;
```

Comment:
I wonder if its worth making this a local type, just since we use it in a few spots.